425 docs rfcadd docs for the mcp server (#4477)
timescale/docs: ai/mcp-server.md

---

> title: Integrate Tiger with your AI Assistant

Reviewer: It's assistant/Assistant 50/50 on this page - can we decide and stick to a case, please?

Author: Assistant

excerpt: Manage your services and optimize your schema and queries using your AI Assistant
products: [cloud, self_hosted]
keywords: [ai, mcp, server, security]
tags: [ai]
---

import RESTPrereqs from "versionContent/_partials/_prereqs-cloud-account-only.mdx";
import CLIINSTALL from "versionContent/_partials/_devops-cli-install.mdx";
import MCPCOMMANDS from "versionContent/_partials/_devops-mcp-commands.mdx";
import MCPCOMMANDSCLI from "versionContent/_partials/_devops-mcp-commands-cli.mdx";
import GLOBALFLAGS from "versionContent/_partials/_devops-cli-global-flags.mdx";

# Integrate Tiger Cloud with your AI Assistant

The $MCP_LONG provides access to your $CLOUD_LONG resources through Claude and other AI Assistants. $MCP_SHORT 
mirrors the functionality of $CLI_LONG and is integrated directly into the $CLI_SHORT binary. You manage your

> $CLOUD_LONG resources including VPCs, services, read replicas, and related infrastructure using natural language 

Reviewer: ```suggestion
$CLOUD_LONG resources including $VPCs, $SERVICE_SHORTs, read replicas, and related infrastructure using natural language 
```

> from your AI Assistant. As $MCP_SHORT is integrated with the $COMPANY documentation, ask any question and you 

Reviewer 2: Worth pointing out that some of these things (VPCs, read replicas) are not technically supported by the MCP server as of right now. But we do plan to implement them in the near future, so maybe fine to leave as-is...? Not sure - your call. Just figured I'd mention it.

Author: Updated

will get the best answer.

This page shows you how to install $CLI_LONG and set up secure authentication for $MCP_SHORT, then manage the
resources in your $ACCOUNT_LONG through the $MCP_LONG using your AI Assistant.

## Prerequisites

<RESTPrereqs />

* An AI Assistant installed on your developer device with an active API key

>   Supported AI Assistants are: `claude-code`, `cursor`, `windsurf`, `codex`, `gemini/gemini-cli`, `vscode/code/vs-code`

Reviewer 2: To be clear, we support any AI assistant that is capable of acting as an MCP client. These are just the common ones that we have easy-to-use install commands for (via `tiger mcp install`).

People can still manually install our MCP server into other AI assistants, though. This is usually done by following that tool's instructions for installing MCP servers, and using `tiger mcp start` as the "command" that gets run (when using the `stdio` protocol, which is most common). See the [Manual Installation](https://github.com/timescale/tiger-cli?tab=readme-ov-file#manual-installation) section of the README. I think it might want to include similar instructions here.

Author: Done

## Install and configure $MCP_SHORT

The $MCP_SHORT is bundled with $CLI_LONG:

<Procedure>

<CLIINSTALL />

> 1. **Configure your AI Assistant to interact with the $PROJECT_SHORTs and $SERVICE_SHORTs in your $ACCOUNT_LONG**

Reviewer: ```suggestion
1. **Configure your AI Assistant to interact with the $PROJECT_SHORT and $SERVICE_SHORTs in your $ACCOUNT_LONG**
```

   For example:
   ```shell
   tiger mcp install claude-code
   ```

>    This command integrates Claude with a local $MCP_SHORT and the $COMPANY documentation.   

Reviewer 2: Worth pointing out that just running `tiger mcp install` (without any other arguments) opens up an interactive prompt where you can choose which client to install it into.

Author: done

</Procedure>

> And that is it, you are ready to use the $MCP_LONG to manage your $SERVICE_SHORTs in $CLOUD_LONG.

Reviewer 2: We may want to also mention that the user needs to log in with `tiger auth login` for the MCP server to work.

Author: This is already mentioned in the <CLIINSTALL /> partial.

## Manage the resources in your $ACCOUNT_LONG through your AI Assistant

Your AI Assistant is connected to your $ACCOUNT_LONG and the $COMPANY documentation, you can now use it to
manage your $SERVICE_SHORTs and learn more about how to implement $CLOUD_LONG features. For example:

<Procedure>

1. **Run your AI Assistant**
   ```shell
   claude
   ```
   Claude automatically runs your $MCP_SHORT server that enables you to interact with $CLOUD_LONG from your
   AI Assistant. 

1. **Check your $MCP_LONG configuration**
   ```shell
   > is the tigerdata mcp server active for you?
   ```
   You see something like:
   ```shell

>    MCP server is active. I can see the following TigerData-related tools available:

Reviewer: ```suggestion
   MCP server is active. I can see the following Tiger Data-related tools available:
```

   - mcp__tiger__get_guide - Retrieve TimescaleDB guides and best practices
   - mcp__tiger__semantic_search_postgres_docs - Search PostgreSQL documentation
   - mcp__tiger__semantic_search_tiger_docs - Search Tiger Cloud and TimescaleDB documentation
   - mcp__tiger__tiger_service_create - Create new database services
   - mcp__tiger__tiger_service_list - List all database services
   - mcp__tiger__tiger_service_show - Show detailed service information
   - mcp__tiger__tiger_service_update_password - Update service passwords

>    Is there something specific you'd like to do with the TigerData MCP server?

Reviewer: ```suggestion
   Is there something specific you'd like to do with the Tiger Data MCP server?
```

   ```

1. **Ask a basic question about your $SERVICE_SHORTs**
   ```shell
   > can you list my active services please
   ```
   You see something like:
   ```shell
   ⏺ You have 3 active services, all with the same name "my-first-service":

   1. tgrservice1 - READY - TimescaleDB - us-east-1 - 1 CPU/4GB - Created: 2025-09-30
   2. tgrservice1 - READY - TimescaleDB - us-east-1 - 1 CPU/4GB - Created: 2025-09-30
   3. tgrservice1 - READY - TimescaleDB - us-east-1 - 1 CPU/4GB - Created: 2025-09-30
   ```

1. **Manage your $SERVICE_SHORTs without having to learn how to**

   For example:
   ```shell

>    Can you create a new ai service called bob with a read replica for me please

Reviewer 2: I'm pretty sure `replicas: 1` creates an HA replica, not a read replica. Maybe just say "a replica" here.

Author: Done

   ```
   You see something like:
   ```shell
    tiger - Create Database Service (MCP)(name: "bob", addons: ["ai"], replicas: 1, wait: false)
   ...
   Do you want to proceed?

>    ❯ 1. Yes 

Reviewer 2: Do we need to include this prompt output? It looks like it only includes the `Yes` option, which is a little weird 😅.

Author: Removed

   ```

1. **Find best practice for things you need to do**

   For example:
   ```shell

>    >  I need to migrate a database with 25 million rows of data to Tiger, 

Reviewer: ```suggestion
   >  I need to migrate a database with 25 million rows of data to Tiger Cloud, 
```

   please tell me what service optimizations i should do before migration. 
   
   ```
   
   You see something like:

   ```shell

>    ⏺ For migrating a database with 25 million rows to Tiger, here are the key service optimizations to do before migration:

Reviewer: ```suggestion
   ⏺ For migrating a database with 25 million rows to Tiger Cloud, here are the key service optimizations to do before migration:
```

   Pre-Migration Service Optimizations

   1. Right-size your compute resources:
     - For 25M rows, start with at least 2 CPU/8GB or 4 CPU/16GB
     - You can scale up during migration if needed, then scale down after

>    2. Enable the appropriate add-ons:

Reviewer: ```suggestion
   2. Enable the appropriate addons:
```

   - Use time-series addon if you're working with time-series data
   - Use ai addon if you need vector/AI capabilities

   3. Choose optimal region:
   - Select the region closest to your current database to minimize network transfer time
   - This significantly reduces migration duration

   4. Consider enhanced storage (Enterprise only):
   - If you expect high write throughput during migration, enhanced storage provides better IOPS (up to 32,000)

   5. Skip replicas initially:
   - Start with 0 replicas during migration
   - Add replicas after migration completes to avoid duplicate data copying

   6. Plan for storage space:
   - Estimate your uncompressed data size
   - Allocate sufficient high-performance storage (scales automatically but good to plan)
   - You can enable tiered storage later for older data

   During Migration Best Practices

   Migration method selection:
   - 25M rows: Use schema-then-data migration (migrate schema first, then copy data in chunks)
     - Allows restart if a copy operation fails
     - Better for databases >10GB

   Database configuration to set AFTER migration:

>    - Set up compression policies (once data is loaded)

Reviewer: Are bullets here and in the Connection strings section supposed to be on different levels?

Author: Nope.
